coral/src/app/features/browse-topics/components/select-team/SelectTeam.tsx

```diff
@@ -1,31 +1,76 @@
 import { NativeSelect, Option } from "@aivenio/design-system";
-import { ChangeEvent } from "react";
+import { useEffect, useState } from "react";
+import { useGetTeams } from "src/app/features/browse-topics/hooks/teams/useGetTeams";
+import { Team } from "src/domain/team";
+import { useSearchParams } from "react-router-dom";
+import { ALL_TEAMS_VALUE } from "src/domain/team/team-types";
 
 type SelectTeamProps = {
-  teams: Array<{ label: string; value: string }>;
-  activeOption: string;
-  selectTeam: (value: string) => void;
+  updateTeam: (teamName: Team | null) => void;
 };
+
 function SelectTeam(props: SelectTeamProps) {
-  const { teams, activeOption, selectTeam } = props;
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialTeam = searchParams.get("team");
+  const [team, setTeam] = useState<Team>(ALL_TEAMS_VALUE);
+  const { updateTeam } = props;
+
+  const { data: topicTeams } = useGetTeams();
+
+  useEffect(() => {
+    if (initialTeam) {
+      setTeam(initialTeam);
+    }
+    // updates `team` in BrowseTopics
+    // which will trigger the api call
+    updateTeam(initialTeam || ALL_TEAMS_VALUE);
+  }, [initialTeam]);
 
-  function onChangeEnv(event: ChangeEvent<HTMLSelectElement>) {
-    selectTeam(event.target.value);
+  function onChangeEnv(newTeam: string) {
+    const isAllTeams = newTeam === ALL_TEAMS_VALUE;
+    if (isAllTeams) {
+      searchParams.delete("team");
+    } else {
+      searchParams.set("team", newTeam);
+    }
+    setTeam(newTeam);
+    updateTeam(newTeam);
+    setSearchParams(searchParams);
   }
 
-  return (
-    <NativeSelect
-      labelText="Team"
-      value={activeOption}
-      onChange={(event) => onChangeEnv(event)}
-    >
-      {teams.map(({ label, value }) => (
-        <Option key={value} value={value}>
-          {label}
+  function createOptions(teams: Team[]) {
```

Comment:
I think this would be readable also inline.

Reply:
Will move it! That was a leftover before I had the check for `topicTeams` and `team` 💡